File: osb-core-main/Api/Osb.Core.Api.Application/Models/Request/GenerateStaticPixQRCodeRequest.cs
using Osb.Core.Platform.Common.Entity.Enums;
using Osb.Core.Platform.Integration.Entity.FitBank.Models;

namespace Osb.Core.Api.Application.Models.Request
{
    public class GenerateStaticPixQRCodeRequest : BaseRequest
    {
        public decimal? PrincipalValue { get; set; }
        public string PixKey { get; set; }
        public PixAddress Address { get; set; }
        public string AdditionalData { get; set; }
        public PixTransactionPurpose? PixTransactionPurpose { get; set; }
        public PixKeyType PixKeyType { get; set; }
    }
}

File: src/features/pix/redux/models/request/generateStaticPixQrCodeRequest.ts
import { ApiRequest } from '_config'
import { PixKeyType } from '../pixKeyType'
import { PixTransactionPurpose } from '../response/enum'
import { PixAddress } from '../response/PixAddress'

export interface GenerateStaticPixQRCodeRequest extends ApiRequest {
  pixKey: string
  principalValue?: number
  address?: PixAddress
  pixTransactionPurpose?: PixTransactionPurpose
  additionalData?: string
  pixKeyType: PixKeyType
}
